### PR TITLE
Bugfix/controller throw

### PR DESCRIPTION
**Pull request by mangoliprasanna**

### CODE REVIEW

File: src/models/token.model.js [hunk 1]
```diff
@@ -45,10 +45,22 @@ const tokenSchema = mongoose.Schema(
...
 };
 
+/**
+ * Create and save access token
+ * @param {object} tokenConfig - javscript object with all required parameters to create token
+ * @param {function(err, model)} cb - Callback function
+ * @returns {Promise<Token>} -
+ */
+tokenSchema.statics.findToken = async function (tokenConfig, cb) {
+  const token = await this.findOne(tokenConfig);
```
Comment (magcentre) [inline src/models/token.model.js]: Didnt we talk about that we will use promises? Why do i see await. Also i see some misunderstanding of promises

File: src/models/user.model.js [hunk 2]
```diff
@@ -106,12 +107,24 @@ userSchema.pre('save', async function (next) {
...
 };
 
+/**
+ * seach user by email address and return
+ * @param {string} id - Users mongoId
+ * @param {Object}  params - The json config of the update values
+ * @returns {Promise<User>}
+ */
+userSchema.statics.getUserByEmail = async function (email) {
+  const user = await this.findOne({ email });
```
Comment (magcentre) [inline src/models/user.model.js]: Same here, moongoose functions are promises by default you can just return the promises,

File: src/processors/user.processor.js [hunk 3]
```diff
@@ -62,18 +83,27 @@ const getUserById = (id) => model.findById(id, { password: 0 });
...
  * @param {string} type
  * @returns {Promise<Token>}
  */
-const verifyToken = (refreshToken) => {
-  const payload = jwt.verify(refreshToken, config.jwt.secret);
-  return token.model.findOne({
-    token: refreshToken, type: token.types.REFRESH, user: payload.sub, blacklisted: false,
-  });
-};
+const verifyToken = (refreshToken) => new Promise((resolve, reject) => {
```
Comment (magcentre) [inline src/processors/user.processor.js]: If you did correctly in models then you wont have to create a promise here.

File: src/processors/user.processor.js [hunk 4]
```diff
@@ -62,18 +83,27 @@ const getUserById = (id) => model.findById(id, { password: 0 });
...
+  utils.verifyJWTToken(refreshToken, config.jwt.secret)
+    .then((decoded) => token.model.findToken({
```
Comment (magcentre) [inline src/processors/user.processor.js]: This should move to model, all model queries should be in model queries

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/swagger.yaml [hunk 5]
```diff
@@ -5,7 +5,7 @@ info:
   description: The micro-service intended to provide the user role and access management
   contact:
     email: prasanna@makonlinesolutions.com
-host: 185.213.175.212:5004
+host: localhost:5004
```
Comment (magcentre) [inline src/swagger.yaml]: Can we remove it

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/middlewares/auth.mw.js [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
 
-    if (!decoded) return sendError(`Authentication failed,: ${info}`, res, 401, req);
+module.exports.jwt = async (req, res, next) => {
+  if (!req.headers.authorization) return sendError({ message: 'Authentication required' }, res, 401, req);
 
-    req.auth = decoded;
+  const token = req.headers.authorization.replace('Bearer ', '');
 
+  try {
+    const decoded = await utils.verifyJWTToken(token, config.jwt.secret);
```
Comment (magcentre) [inline src/middlewares/auth.mw.js]: I still see await

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/controllers/user.controller.js [hunk 7]
```diff
@@ -7,34 +7,24 @@ const processor = require('../processors/user.processor');
...
-  processor.getUserByEmail(loginBody.email)
-    .then((user) => {
-      if (user) return user.isPasswordMatch(loginBody.password);
-      sendError('Unregisted email address', res, 400, req);
-    })
-    .then((user) => {
-      if (user.match) return processor.generateAndSaveAuthToken(user);
-      sendError('Invalid email and password', res, 400, req);
-    })
+  processor.verifyEmailAndPassword(loginBody.email, loginBody.password)
```
Comment (magcentre) [inline src/controllers/user.controller.js]: Only one processor function should be called. The second processor should be called inside the first processor function. 

Remember chaining of processors is business logic and we write busin ss logic in processors

File: src/controllers/user.controller.js [hunk 8]
```diff
@@ -71,7 +57,7 @@ const getAccessToken = (req, res) => {
 const updateProfile = (req, res) => {
   const userBody = req.body;
 
-  processor.isEmailTaken(userBody.email, req.auth.sub)
+  processor.verifyEmail(userBody.email, req.auth.sub)
```
Comment (magcentre) [inline src/controllers/user.controller.js]: Same here. Processor logic should go inside the processors

**Review by ghost — CHANGES_REQUESTED** [review]

**Pull request merged**